src/app/hooks/useWeather.ts

```diff
@@ -0,0 +1,39 @@
+import useFetch from './useFetch';
+import { useEffect, useState } from 'react';
+import getCoords from '../utils/getCoords';
+
+type useWeatherType = {
+  degree: number | undefined;
+  icon: string | undefined;
+  description: string | undefined;
+  wind: number | null | undefined;
+};
+
+export default function useWeather(): useWeatherType {
+  const [location, setLocation] = useState<{
+    lat?: number | null;
+    long?: number | null;
+  } | null>({ lat: 0, long: 0 });
+
+  useEffect(() => {
+    async function run() {
+      const location = await getCoords();
```

Comment:
But I always click "block" when pages ask me for my location :)

Reply:
But If you want to use a weather App you should allow it, I guess?

Reply:
Pls don´t click 'block"  :-D